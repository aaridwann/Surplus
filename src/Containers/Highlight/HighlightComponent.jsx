// import React from 'react';
// import { FlatList, Image, Text, View } from 'react-native';
// import { scaleFont, scaleHeight, scaleSize } from '../../Utils/Scale/Scale';
// import Constants from '../../Utils/Constants/Constants';

// const renderItem = ({ item }) => (
// 	<View style={{ marginHorizontal: 8 }}>
// 		<Image
// 			source={{ uri: item?.picture.medium }}
// 			style={{
// 				borderRadius: scaleHeight(50) / 2,
// 				backgroundColor: Constants.Colors.LightGreenSurplus,
// 				width: scaleHeight(50),
// 				height: scaleHeight(50),
// 			}}
// 		/>
// 		<Text style={{ fontSize: scaleFont(15), marginVertical: 4 }}>
// 			{item?.name?.first}
// 		</Text>
// 	</View>
// );

// function HighlightComponent({ data }) {
// 	return (
// 		<View
// 			style={{
// 				minHeight: scaleHeight(80),
// 				paddingVertical: scaleSize(10),
// 				backgroundColor: 'white',
// 				width: '100%',
// 				padding: scaleSize(8),
// 			}}
// 		>
// 			<Text style={{ fontSize: scaleSize(15) }}>Highlight</Text>
// 			<FlatList
// 				contentContainerStyle={{ marginTop: 10 }}
// 				horizontal={true}
// 				data={data}
// 				renderItem={renderItem}
// 				keyExtractor={(item, i) => i}
// 			/>
// 		</View>
// 	);
// }

// export default HighlightComponent;
import React from 'react';
import PropTypes from 'prop-types';
import { FlatList, Image, StyleSheet, Text, View } from 'react-native';
import { scaleFont, scaleHeight, scaleSize } from '../../Utils/Scale/Scale';
import Constants from '../../Utils/Constants/Constants';

const styles = StyleSheet.create({
	container: {
		minHeight: scaleHeight(80),
		paddingVertical: scaleSize(10),
		backgroundColor: 'white',
		width: '100%',
		padding: scaleSize(8),
	},
	title: {
		fontSize: scaleSize(15),
	},
	itemContainer: {
		marginHorizontal: 8,
	},
	image: {
		borderRadius: scaleHeight(50) / 2,
		backgroundColor: Constants.Colors.LightGreenSurplus,
		width: scaleHeight(50),
		height: scaleHeight(50),
	},
	text: {
		fontSize: scaleFont(15),
		marginVertical: 4,
	},
});

const renderItem = ({ item, index }) => (
	<View testID={`highlight-item-${index}`} style={styles.itemContainer}>
		<Image
			source={{ uri: item?.picture.medium }}
			style={styles.image}
			testID={`highlight-image-${index}`}
		/>
		<Text style={styles.text} testID={`highlight-text-${index}`}>
			{item?.name?.first}
		</Text>
	</View>
);

function HighlightComponent({ data }) {
	return (
		<View style={styles.container} testID="highlight-component">
			<Text style={styles.title}>Highlight</Text>
			<FlatList
				contentContainerStyle={{ marginTop: 10 }}
				horizontal={true}
				data={data}
				renderItem={renderItem}
				keyExtractor={(item, index) => `highlight-item-${index}`}
			/>
		</View>
	);
}

HighlightComponent.propTypes = {
	data: PropTypes.arrayOf(
		PropTypes.shape({
			picture: PropTypes.shape({
				medium: PropTypes.string.isRequired,
			}).isRequired,
			name: PropTypes.shape({
				first: PropTypes.string.isRequired,
			}).isRequired,
		})
	).isRequired,
};

export default HighlightComponent;
