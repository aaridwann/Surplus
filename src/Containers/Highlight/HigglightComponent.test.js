import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import HighlightComponent from './HighlightComponent';

const data = [
	{
		picture: {
			medium: 'https://example.com/image1.jpg',
		},
		name: {
			first: 'John',
		},
	},
	{
		picture: {
			medium: 'https://example.com/image2.jpg',
		},
		name: {
			first: 'Jane',
		},
	},
];

describe('HighlightComponent', () => {
	it('should render correctly', () => {
		const { toJSON } = render(<HighlightComponent data={data} />);
		expect(toJSON()).toMatchSnapshot();
	});

	it('should render the correct number of items', () => {
		const { getAllByTestId } = render(<HighlightComponent data={data} />);
		const items = getAllByTestId(/^highlight-item-/);
		expect(items).toHaveLength(2);
	});

	// it('should call onPress when an item is pressed', () => {
	// 	const onPress = jest.fn();
	// 	const { getByTestId } = render(
	// 		<HighlightComponent data={data} onPress={onPress} />
	// 	);

	// 	const item = getByTestId('highlight-item-0');
	// 	fireEvent.press(item);
	// 	expect(onPress).toHaveBeenCalledTimes(1);
	// 	expect(onPress).toHaveBeenCalledWith(data[0]);
	// });
});
