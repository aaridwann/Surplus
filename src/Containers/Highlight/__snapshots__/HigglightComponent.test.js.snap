// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HighlightComponent should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "minHeight": 160,
      "padding": 16,
      "paddingVertical": 20,
      "width": "100%",
    }
  }
  testID="highlight-component"
>
  <Text
    style={
      Object {
        "fontSize": 30,
      }
    }
  >
    Highlight
  </Text>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "marginTop": 10,
      }
    }
    data={
      Array [
        Object {
          "name": Object {
            "first": "John",
          },
          "picture": Object {
            "medium": "https://example.com/image1.jpg",
          },
        },
        Object {
          "name": Object {
            "first": "Jane",
          },
          "picture": Object {
            "medium": "https://example.com/image2.jpg",
          },
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          style={
            Object {
              "marginHorizontal": 8,
            }
          }
          testID="highlight-item-0"
        >
          <Image
            source={
              Object {
                "uri": "https://example.com/image1.jpg",
              }
            }
            style={
              Object {
                "backgroundColor": "#50b8a8",
                "borderRadius": 50,
                "height": 100,
                "width": 100,
              }
            }
            testID="highlight-image-0"
          />
          <Text
            style={
              Object {
                "fontSize": 30,
                "marginVertical": 4,
              }
            }
            testID="highlight-text-0"
          >
            John
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          style={
            Object {
              "marginHorizontal": 8,
            }
          }
          testID="highlight-item-1"
        >
          <Image
            source={
              Object {
                "uri": "https://example.com/image2.jpg",
              }
            }
            style={
              Object {
                "backgroundColor": "#50b8a8",
                "borderRadius": 50,
                "height": 100,
                "width": 100,
              }
            }
            testID="highlight-image-1"
          />
          <Text
            style={
              Object {
                "fontSize": 30,
                "marginVertical": 4,
              }
            }
            testID="highlight-text-1"
          >
            Jane
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
